doc/ssl/SSL_CTX_set_ct_validation_callback.pod

```diff
@@ -2,39 +2,89 @@
 
 =head1 NAME
 
+SSL_ct_enable, SSL_CTX_ct_enable, SSL_ct_disable, SSL_CTX_ct_disable,
 SSL_set_ct_validation_callback, SSL_CTX_set_ct_validation_callback,
-SSL_get_ct_validation_callback, SSL_CTX_get_ct_validation_callback -
+SSL_ct_is_enabled, SSL_CTX_ct_is_enabled -
 control Certificate Transparency policy
 
 =head1 SYNOPSIS
 
  #include <openssl/ssl.h>
 
- int SSL_set_ct_validation_callback(SSL *s, ct_validation_cb callback, void *arg);
- int SSL_CTX_set_ct_validation_callback(SSL_CTX *ctx, ct_validation_cb callback, void *arg);
- ct_validation_cb SSL_get_ct_validation_callback(const SSL *s);
- ct_validation_cb SSL_CTX_get_ct_validation_callback(const SSL_CTX *ctx);
+ int SSL_ct_enable(SSL *s, int vtype);
+ int SSL_CTX_ct_enable(SSL_CTX *ctx, int vtype);
+ int SSL_set_ct_validation_callback(SSL *s, ssl_ct_validation_cb callback,
+                                    void *arg);
+ int SSL_CTX_set_ct_validation_callback(SSL_CTX *ctx,
+                                        ssl_ct_validation_cb callback,
+                                        void *arg);
+ void SSL_ct_disable(SSL *s);
+ void SSL_CTX_ct_disable(SSL_CTX *ctx);
+ int SSL_ct_is_enabled(const SSL *s);
+ int SSL_CTX_ct_is_enabled(const SSL_CTX *ctx);
 
 =head1 DESCRIPTION
 
-SSL_set_ct_validation_callback() and SSL_CTX_set_ct_validation_callback() set
-the function that is called when Certificate Transparency validation needs to
-occur. It is the responsibility of this function to examine the signed
-certificate timestamps (SCTs) that are passed to it and determine whether they
-are sufficient to allow the connection to continue. If they are, the function
-must return 1, otherwise it must return 0.
-
-An arbitrary piece of user data, B<arg>, can be passed in when setting the
-callback. This will be passed to the callback whenever it is invoked. Ownership
-of this userdata remains with the caller.
+SSL_ct_enable() and SSL_CTX_ct_enable() enable the processing of signed
+certificate timestamps (SCTs) either for a given SSL connection or for all
+connections that share the given SSL context, respectively.
+This is accomplished by setting a built-in CT validation callback.
+The behaviour of the callback is determined by the B<vtype> argument.
+
+If B<vtype> is equal to B<SSL_CT_VALIDATION_PERMISSIVE>, then the
```

Comment:
We should probably finish our "enums considered helpful" vs "enums considered harmful" argument... Particularly since the rest of CT lib prefers enums over defines (also in public interfaces).

Rob, in case Ben didn't relay this to you, the concern is that enums aren't guaranteed backwards-compatible when you recompile the library (compiler can choose type/width arbitrarily).

Option A is to

```
enum {
SSL_CT_VALIDATION_PERMISSIVE = -1,
/* More enums here */
};

/* You can now use ct_version_t in your interface but it's typedef'ed to a plain int. */
typedef int ct_version_t;
```

Option B is to assume that compilers behave sensibly.

Either way, @vdukhovni, I think you should either convince Rob to change all the other ones (option A), or follow the existing style in ct lib (option B).